resources/assets/lib/utils/beatmap-helper.ts

```diff
@@ -18,6 +19,12 @@ function isVisibleBeatmap(beatmap: BeatmapJson) {
   return true;
 }
 
+const difficultyColourSpectrum = d3.scaleLinear<string>()
+  .domain([1.5, 2, 2.5, 3.375, 4.625, 5.875, 7, 8])
```

Comment:
where did these numbers come from? The colours match the gradient but the numbers don't seem to map &times;8 of the percentages? (and why calculated manually)

```
background: linear-gradient(180deg, #4FC0FF 21.59%, #4FFFD5 26.93%, #7CFF4F 32.17%, #F7F05C 41.11%, #FF8068 57.67%, #FF3C71 73.07%, #6563DE 86.67%, #18158E 99.96%), #59FFE9;
```